tests/mutation_test.cc

```diff
@@ -1320,6 +1320,99 @@ SEASTAR_THREAD_TEST_CASE(test_mutation_upgrade_type_change) {
     assert_that(m).is_equal_to(m2);
 }
 
+SEASTAR_THREAD_TEST_CASE(test_row_marker) {
+    auto ttl = std::chrono::seconds(1);
+    auto t1 = gc_clock::now();
+    auto t0 = t1 - std::chrono::seconds(1);
+    auto t2 = t1 + std::chrono::seconds(1);
+    auto t3 = t2 + std::chrono::seconds(1);
+
+    can_gc_fn never_gc = [] (tombstone) { return false; };
+
+    auto must_be_alive = [&] (row_marker mark, gc_clock::time_point t) {
+        BOOST_TEST_MESSAGE(format("must_be_alive({}, {})", mark, t));
+        BOOST_REQUIRE(mark.is_live(tombstone(), t));
+        BOOST_REQUIRE(mark.is_missing() || !mark.is_dead(t));
+        BOOST_REQUIRE(mark.compact_and_expire(tombstone(), t, never_gc, gc_clock::time_point()));
+    };
+
+    auto must_be_dead = [&] (row_marker mark, gc_clock::time_point t) {
+        BOOST_TEST_MESSAGE(format("must_be_dead({}, {})", mark, t));
+        BOOST_REQUIRE(!mark.is_live(tombstone(), t));
+        BOOST_REQUIRE(mark.is_missing() || mark.is_dead(t));
+        BOOST_REQUIRE(!mark.compact_and_expire(tombstone(), t, never_gc, gc_clock::time_point()));
+    };
+
+    // Without timestamp it is never alive
```

Comment:
Without timestamp the marker is missing (doesn't exist).

Reply:
Changed.